### PR TITLE
Feature/3254 settings UI ga4

### PR DESCRIPTION
**Pull request by ivankruchkoff**
## Summary

Addresses issue #3254

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by asvinb** [comment]
@googlebot I consent.

File: assets/js/modules/analytics/components/settings/SettingsView.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/settings/SettingsView.js:41]: We should rely on `const isGA4Enabled = useFeature( 'ga4setup' );` when we interact with the `analytics-4` module functionality. In other words we should try to pull `propertyID` only if the `isGA4Enabled` is true.

File: assets/js/modules/analytics/components/settings/SettingsForm.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/settings/SettingsForm.js:58]: Let's use `useFeature` hook instead:
```suggestion
	const ga4Enabled = useFeature( 'ga4setup' );
```

File: assets/js/modules/analytics/components/settings/SettingsForm.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/settings/SettingsForm.js:42]:
```suggestion
import PropertySelect4 from '../../../analytics-4/components/common/PropertySelect';
```

File: assets/js/modules/analytics/components/settings/SettingsEdit.test.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/settings/SettingsEdit.test.js:37]: The `status` is already `200` by default, so we can omit it:
```suggestion
			{ body: [] },
```

File: assets/js/modules/analytics/components/settings/SettingsForm.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/settings/SettingsForm.js:69]: It should be done in the `useEffect` hook.

File: assets/js/modules/analytics/components/settings/SettingsForm.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/settings/SettingsForm.js:64]: We should do it only if the `ga4Enabled` is true.

File: assets/js/modules/analytics/components/settings/SettingsForm.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/settings/SettingsForm.js:88]:
```suggestion
				( ga4Enabled && SETUP_FLOW_MODE_GA4_TRANSITIONAL === setupFlowMode ) && (
```

File: assets/js/modules/analytics/components/settings/SettingsForm.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/settings/SettingsForm.js:91]:
```suggestion
							{ __( 'You’ll need to connect the Google Analytics 4 property that’s associated with this Universal Analytics property', 'google-site-kit' ) }
```

File: assets/js/modules/analytics/components/settings/SettingsForm.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/settings/SettingsForm.js:107]:
```suggestion
				( ga4Enabled && SETUP_FLOW_MODE_UA === setupFlowMode ) && (
```

**Review by eugene-manuilov — DISMISSED** [review]
Thanks, @ivankruchkoff. It's a good start, however, there are some things that we need to improve before we can approve this PR. Please, take a look at my comments.

**Comment by eugene-manuilov** [comment]
@googlebot I consent.

File: assets/js/modules/analytics/components/settings/SettingsForm.js
Comment (tofumatt) [inline assets/js/modules/analytics/components/settings/SettingsForm.js:65]: This seems a bit of a hack—relying on the side effect of a selector to call another one.

It makes me think we should be calling these selectors as part of `getSetupFlowMode`'s resolver.

Why can't we do that here?

**Review by tofumatt — CHANGES_REQUESTED** [review]
This looks fine apart from one thing: relying on a selector's side-effect before calling another one.

Having to manage that flow in a component seems the wrong approach, as it's entirely what resolvers are for. Is there any reason we _can't_ call:

```js
const accountID = select( STORE_NAME ).getAccountID();
select( MODULES_ANALYTICS_4 ).getProperties( accountID );
```

in the resolver for `getSetupFlow`? Or even as a `createRegistrySelector` in part of its selector?

File: assets/js/modules/analytics/components/settings/SettingsForm.js
Comment (ivankruchkoff) [reply assets/js/modules/analytics/components/settings/SettingsForm.js:65]: My initial idea was to change that resolver, however it wasn't in the IB, otherwise this hack has to happen everywhere we call the selector.

Comment (tofumatt) [reply assets/js/modules/analytics/components/settings/SettingsForm.js:65]: As in: if we don't change the resolver this needs to be added everywhere we call the `getSetupFlowMode`'s resolver?

That's a recipe for bugs in my mind; I see it isn't in the IB, but let's modify the resolver/selector to call these dependant selectors first.

----

Actually, I'm a bit confused; I went to look at the selector for `getSetupFlowMode` and it already calls the selectors you're looking for: https://github.com/google/site-kit-wp/blob/922999e8bb12bb40108244e992b61e5c9c07f88f/assets/js/modules/analytics/datastore/setup-flow.js#L56-L74

Comment (eugene-manuilov) [reply assets/js/modules/analytics/components/settings/SettingsForm.js:65]: > Actually, I'm a bit confused; I went to look at the selector for getSetupFlowMode and it already calls the selectors you're looking for:

The `getProperties` may return undefined because it has to be resolved first. That's why we call it beforehand.

Comment (tofumatt) [reply assets/js/modules/analytics/components/settings/SettingsForm.js:65]: Once it's resolved the selectors should all update though, no?

We account for that here: https://github.com/google/site-kit-wp/blob/b252720322e3dce7883c92c79811c6d83e19f20c/assets/js/modules/analytics/datastore/setup-flow.js#L76-L78

If `undefined` is returned by `getSetupFlowMode` we should render a loading screen/null/something else until things have loaded. The selector is designed to work without needing things to resolve first, I think here we just have to account for it returning `undefined` while things load.

**Review by tofumatt — CHANGES_REQUESTED** [review]
I think we need to adjust `SettingsForm` to allow to `getSetupFlowMode()` to return `undefined` while loading. That's part of the selector's design and I don't think we should be relying on components to call selectors in order like that.

File: assets/js/modules/analytics/components/settings/SettingsForm.js
Comment (ivankruchkoff) [reply assets/js/modules/analytics/components/settings/SettingsForm.js:65]: `isAdminAPIWorking()` returns undefined without it
https://github.com/google/site-kit-wp/blob/922999e8bb12bb40108244e992b61e5c9c07f88f/assets/js/modules/analytics-4/datastore/api.js#L32

which gets us to here:
https://github.com/google/site-kit-wp/blob/b252720322e3dce7883c92c79811c6d83e19f20c/assets/js/modules/analytics/datastore/setup-flow.js#L49

Comment (eugene-manuilov) [reply assets/js/modules/analytics/components/settings/SettingsForm.js:65]: @tofumatt let's create a follow-up ticket for this if you think it's worth it and not block this PR because it is not relevant to this ticket.

**Review by tofumatt — APPROVED** [review]
In light of the issues with the `getSetupFlowMode` selector this looks good; I'll open a follow-up PR for that selector.

Thanks!

**Pull request merged**
